src/main/kotlin/Network.kt

```diff
@@ -11,14 +11,22 @@ class Network(val capacity: Int, val listOfCars: MutableList<Car>) {
     }
 
     fun analyzeNetwork() {
-        val requiredCapacity = this.requiredCapacity()
-        val carsDividedByCapacity = requiredCapacity / capacity.toDouble()
-        val chanceOfDelay = chanceOfDelay(carsDividedByCapacity)
+        val hours = 24
+        for (hour in 1..hours) {
+            val requiredCapacity = this.requiredCapacity()
+            val carsDividedByCapacity = requiredCapacity / capacity.toDouble()
+            val chanceOfDelay = chanceOfDelay(carsDividedByCapacity)
 
-        for (car in listOfCars) {
-            if (car.wantsToDrive) {
-                val delayed = computingDelay(chanceOfDelay)
-                car.isDelayed = delayed
+            for (car in listOfCars) {
+                if (car.wantsToDrive) {
```

Comment:
You might want to keep `wantsToDrive` around to keep CSV parsing working but you shouldn't use it anymore because it does not know about 24 hours.

I recommend to have a function `wantsToDrive(hour: Int)` on `Car` that returns whether a car wants to drive at that hour. `Network` should then call it.